packages/back-end/src/models/DataSourceModel.ts

```diff
@@ -205,3 +207,10 @@ export async function _dangerousGetAllDatasources(): Promise<
   const all = await DataSourceModel.find();
   return all.map(toInterface);
 }
+
+async function onDataSourceCreate(
+  datasource: DataSourceInterface,
+  organization: string
+) {
+  await createInitialInformationSchema(datasource, organization);
```

Comment:
Currently, if we are unable to generate the informationSchema for a particular datasource, it fails silently and doesn't impede the creation of the datasource as a whole.

In follow-up PR's I plan to create a POST and a PUT endpoint that will handle creating/updating a datasource's informationSchema. This will primarily be consumed by a button that allows a user to generate an informationSchema if one does not exist (and the datasource type is supported) or update an existing informationSchema. In both of those cases, I will build logic in so if the informationSchema isn't successfully generated we return a contextual error.

Let me know if you think that is necessary here.

Reply:
Makes sense. Just one thought- the 'fail silently' logic can be encapsulated here by leveraging `try / catch` when calling `createInitialInformationSchema`. My thought is it would read better that way, by being obvious to the reader that inside `onDataSourceCreate`, "we want non-blocking things to happen".

This would also clean up the implementation of `createInitialInformationSchema` and `generateInformationSchema` a little bit by not having to be concerned (as much) with error-handling, if that makes sense.